source/opt/basic_block.h

```diff
@@ -62,19 +62,19 @@ class BasicBlock {
   const_iterator cbegin() { return const_iterator(&insts_, insts_.cbegin()); }
   const_iterator cend() { return const_iterator(&insts_, insts_.cend()); }
 
-  // Runs the given function |f| on each instruction in this basic block.
-  inline void ForEachInst(const std::function<void(Instruction*)>& f);
-
-  // Pushes the binary segments for this instruction into the back of *|binary|.
-  // If |skip_nop| is true and this is a OpNop, do nothing.
-  inline void ToBinary(std::vector<uint32_t>* binary, bool skip_nop) const;
+  // Runs the given function |f| on each instruction in this basic block, and
+  // optionally on the debug line instructions that might precede them.
+  inline void ForEachInst(const std::function<void(Instruction*)>& f,
+                          bool run_on_debug_line_insts = false);
+  inline void ForEachInst(const std::function<void(const Instruction*)>& f,
```

Comment:
Maybe we should change the parameter to `const std::function<void(const Instruction&)>` so that we don't need those `static_cast` to aid compilers to do resolution? This also satisfies the style guide.